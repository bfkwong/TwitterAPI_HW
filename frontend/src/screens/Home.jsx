import { Grid, Container, Typography, IconButton } from "@material-ui/core";
import React, { useEffect, useState } from "react";
import Tweet from "../components/Tweet";
import Tweeting from "../components/Tweeting";
import ExitToAppIcon from "@material-ui/icons/ExitToApp";
import { Redirect, useHistory } from "react-router";
import { getAllTweets } from "../Requests";

export default function Home(props) {
  const [allTweets, setAllTweets] = useState([]);
  const history = useHistory();

  const requestTweets = async () => {
    const allTweetsResp = await getAllTweets();
    setAllTweets(allTweetsResp);
  };

  useEffect(() => {
    requestTweets();
  }, []);

  if (!props.auth) {
    return <Redirect to="/login" />;
  }

  return (
    <React.Fragment>
      <Container maxWidth="sm" style={{ marginTop: 30 }}>
        <Grid container spacing={3}>
          <Grid item xs={12}>
            <Grid container alignItem="center">
              <Grid item xs>
                <Typography variant="h5">Fake Twitter</Typography>
              </Grid>
              <Grid item>
                <IconButton
                  onClick={async () => {
                    await props.setAuth(false);
                    history.push("/login");
                  }}>
                  <ExitToAppIcon />
                </IconButton>
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={12}>
            <Tweeting onTweet={requestTweets} />
          </Grid>
          {allTweets?.length > 0 &&
            allTweets.map((tweetData) => (
              <Grid item xs={12}>
                <Tweet tweet={tweetData} onDelete={requestTweets} />
              </Grid>
            ))}
        </Grid>
      </Container>
    </React.Fragment>
  );
}
