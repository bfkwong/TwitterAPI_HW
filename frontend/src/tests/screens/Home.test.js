import Home from "../../screens/Home";
import { render } from "@testing-library/react";

test("Tweet with prop should show correctly", async () => {
  const setAuthMock = jest.fn();
  const component = render(<Home auth={true} setAuth={setAuthMock} />);

  expect(component).toMatchSnapshot();
});
