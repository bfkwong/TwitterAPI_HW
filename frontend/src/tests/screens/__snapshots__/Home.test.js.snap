// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet with prop should show correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
        style="margin-top: 30px;"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              alignitem="center"
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5"
                >
                  Fake Twitter
                </h5>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                  >
                    <textarea
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                      placeholder="What's on your mind?"
                      rows="4"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-2"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-contained"
                  >
                    280 characters left
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-flex-end"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Tweet
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm"
      style="margin-top: 30px;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            alignitem="center"
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                Fake Twitter
              </h5>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    placeholder="What's on your mind?"
                    rows="4"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-2"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained"
                >
                  280 characters left
                </p>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-flex-end"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Tweet
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
